sf_examples/vizdoom/doom/multiplayer/doom_multiagent_wrapper.py

```diff
@@ -217,13 +218,22 @@ def __init__(self, num_agents, make_env_func, env_config, skip_frames, render_mo
     def get_default_reward_shaping(self):
         return self.default_reward_shaping
 
-    def set_reward_shaping(self, reward_shaping: dict, agent_idx: int):
-        self.current_reward_shaping[agent_idx] = reward_shaping
-        self.set_env_attr(
-            agent_idx,
-            "unwrapped.reward_shaping_interface.reward_shaping_scheme",
-            reward_shaping,
-        )
+    def set_reward_shaping(self, reward_shaping: dict, agent_indices: Union[int, slice]):
+        if isinstance(agent_indices, slice):
```

Comment:
You're repeating the same code here twice 
If `agent_indices` is not a slice just convert it to a slice of lenght 1 and use the same for-loop code! :)